CRM/Activity/Form/Task/SMS.php

```diff
@@ -41,17 +41,19 @@ class CRM_Activity_Form_Task_SMS extends CRM_Activity_Form_Task {
    */
   public function preProcess() {
     parent::preProcess();
-    $form = $this;
     $this->bounceOnNoActiveProviders();
     $activityCheck = 0;
+    // This is really bad - we are doing a check on a language-specific subject...
+    // shouldn't we be using an activity type instead???
```

Comment:
Well, at least we acknowledge that it's really bad!

Reply:
that's the first part of any apology